libs/server-sdk/src/data_systems/background_sync/detail/payload_filter_validation/payload_filter_validation.cpp

```diff
@@ -0,0 +1,23 @@
+#include "payload_filter_validation.hpp"
+
+#include <boost/regex.hpp>
+
+namespace launchdarkly::server_side::data_systems::detail {
+bool ValidateFilterKey(std::string const& filter_key) {
+    if (filter_key.empty()) {
+        return false;
+    }
+    try {
+        return regex_search(filter_key,
+                            boost::regex("^[a-zA-Z0-9][._\\-a-zA-Z0-9]*$"));
```

Comment:
I think we should try to keep these regexes the same and `\w` more concisely represents `A-Za-z0-9_`.